src/main/java/org/openrewrite/java/spring/boot2/search/IntegrationSchedulerPoolRecipe.java

```diff
@@ -169,9 +169,13 @@ public Properties visitFile(Properties.File file, ExecutionContext context) {
                 return (SourceFile) new JavaIsoVisitor<ExecutionContext>() {
                     @Override
                     public J.Annotation visitAnnotation(J.Annotation annotation, ExecutionContext context) {
-                        List<Comment> comments = new ArrayList<>(annotation.getComments());
-                        comments.add(new TextComment(false, GENERAL_MIGRATION_MESSAGE, "", Markers.EMPTY));
-                        return annotation.withComments(comments);
+                        J.Annotation a = super.visitAnnotation(annotation, context);
+                        if (annotationMatcher.matches(a)) {
+                            List<Comment> comments = new ArrayList<>(a.getComments());
```

Comment:
@BoykoAlex I'd prefer we use `ListUtils.concat` or `ListUtils.concatAll` for these kinds of scenarios, which will allow us to swap out `ArrayList` later if we find a better performing implementation.